build-logic/build.gradle.kts

```
@@ -0,0 +1,20 @@
+plugins {
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenCentral()
+}
+
+// Pass '-PkotlinDev' to command line to enable kotlin-in-development version
+val kotlinVersion = if (project.hasProperty("kotlinDev")) {
+    logger.warn("Enabling kotlin dev version!")
+    libs.versions.kotlinDev.get()
+} else {
+    libs.versions.kotlin.get()
+}
+
+dependencies {
+    implementation(kotlin("gradle-plugin", kotlinVersion))
+    implementation(libs.dokka)
+}
```

This goes against the laziness version catalogs are built on, it's probably better (and simpler using libs):
```suggestion
dependencies {
    if (project.hasProperty("kotlinDev")) {
        // Pass '-PkotlinDev' to command line to enable kotlin-in-development version
        logger.warn("Enabling kotlin dev version!")
        implementation(libs.kotlin.plugin.dev)
    } else {
        implementation(libs.kotlin.plugin)
    }
    implementation(libs.dokka)
}
```

where
```
[libraries]
kotlin-plugin = { module = "org.jetbrains.kotlin:kotlin-gradle-plugin", version.ref = "kotlin" }
kotlin-plugin-dev = { module = "org.jetbrains.kotlin:kotlin-gradle-plugin", version.ref = "kotlinDev" }
```

---

Could also be tweaked to below?
```kotlin
dependencies {
    val kotlinPlugin = if (project.hasProperty("kotlinDev")) {
        // Pass '-PkotlinDev' to command line to enable kotlin-in-development version
        logger.warn("Enabling kotlin dev version!")
        libs.kotlin.plugin.dev
    } else {
        libs.kotlin.plugin
    }
    implementation(kotlinPlugin)
    implementation(libs.dokka)
}
```

---

8670227077af7d7ec25f0335c6359f52fa4a0b73